tests/test_root_commands.py

```diff
@@ -132,12 +141,81 @@ def complete(partialPath):
     assert complete('t') == ['tags.', 'theme.', 'tmp.']
     assert complete('') == [l + '.' for l in hlwm.list_children_via_attr('')]
 
-@pytest.mark.parametrize('attrtype', ['int', 'bool', 'string', 'color', 'uint'])
+
+@pytest.mark.parametrize('attrtype', ATTRIBUTE_TYPES)
 @pytest.mark.parametrize('name', ['my_test', 'my_foo'])
 @pytest.mark.parametrize('object_path', ['', 'clients', 'theme.tiling.active'])
-def test_user_attributes_without_removal(hlwm, attrtype, name, object_path):
+def test_new_attr_without_removal(hlwm, attrtype, name, object_path):
     path = (object_path + '.' + name).lstrip('.')
 
     hlwm.call(['new_attr', attrtype, path])
 
     hlwm.get_attr(path)
+
+
+@pytest.mark.parametrize('attrtype', ATTRIBUTE_TYPES)
+def test_new_attr_existing_attribute(hlwm, attrtype):
+    hlwm.get_attr('monitors.count')
+    hlwm.call_xfail(['new_attr', attrtype, 'monitors.count']) \
+        .match('attribute name must start with "my_"')
+
+
+@pytest.mark.parametrize('attrtype', ATTRIBUTE_TYPES)
+def test_new_attr_existing_user_attribute(hlwm, attrtype):
+    path = 'theme.my_user_attr'
+    hlwm.call(['new_attr', attrtype, path])
+    hlwm.get_attr(path)
+
+    hlwm.call_xfail(['new_attr', attrtype, path]) \
+        .match('already has an attribute')
+
+
+@pytest.mark.parametrize('attrtype', ATTRIBUTE_TYPES)
+@pytest.mark.parametrize('path', ['foo', 'monitors.bar'])
+def test_new_attr_missing_prefix(hlwm, attrtype, path):
+    hlwm.call_xfail(['new_attr', attrtype, path]) \
+        .match('must start with "my_"')
+
+
+@pytest.mark.parametrize('attrtype,values', zip(ATTRIBUTE_TYPES, ATTRIBUTE_VALUES))
+@pytest.mark.parametrize('path', ['my_foo', 'monitors.my_bar'])
+def test_new_attr_writable(hlwm, attrtype, values, path):
+    hlwm.call(['new_attr', attrtype, path])
+    for v in values:
+        hlwm.call(['set_attr', path, v])
+        assert hlwm.get_attr(path) == str(v)
+
+
+@pytest.mark.parametrize('attrtype', ATTRIBUTE_TYPES)
+def test_new_attr_right_type(hlwm, attrtype):
```

Comment:
Just a hint: Don't be afraid to use verbs in unit test names. ;-) Those test names are totally fine, but "test_new_attr_has_right_type" and "test_new_attr_is_writable" would be even better, in my opinion.